File: beautiful matrix A.java
import java.util.Scanner;

public class BeautifulMatrix {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int oneRow = -1;
        int oneCol = -1;
        for (int i = 0; i < 5; i++) {
            for (int j = 0; j < 5; j++) {
                int value = scanner.nextInt();
                if (value == 1) {
                    oneRow = i + 1; 
                    oneCol = j + 1; 
                }
            }
        }

        int targetRow = 3;
        int targetCol = 3;
        int rowMoves = Math.abs(oneRow - targetRow);
        int colMoves = Math.abs(oneCol - targetCol);
  
        int totalMoves = rowMoves + colMoves;
        System.out.println(totalMoves);
        
        scanner.close();
    }
}

File: beautiful matrix A.py
one_row = -1
one_col = -1

for i in range(5):
    row = list(map(int, input().split()))
    for j in range(5):
        value = row[j]
        if value == 1:
            one_row = i + 1
            one_col = j + 1

target_row = 3
target_col = 3
row_moves = abs(one_row - target_row)
col_moves = abs(one_col - target_col)

total_moves = row_moves + col_moves
print(total_moves)
